### ISSUE TITLE
R CMD check igraph_1.2.7.tar.gz failed when test igraph@1.2.7 on centos8_x86_64

### ISSUE THREAD

**Issue by Tom-python0121** [igraph/rigraph#487]
### Steps to reproduce the issue

```console
[root@centos8 spack-stage-r-igraph-1.2.7-dmnjavnjyqtow5b5r4ip5im66ux323zj]# R CMD check igraph_1.2.7.tar.gz
* using log directory ‘/tmp/root/spack-stage/spack-stage-r-igraph-1.2.7-dmnjavnjyqtow5b5r4ip5im66ux323zj/igraph.Rcheck’
* using R version 4.0.4 (2021-02-15)
* using platform: aarch64-unknown-linux-gnu (64-bit)
* using session charset: UTF-8
* checking for file ‘igraph/DESCRIPTION’ ... OK
* this is package ‘igraph’ version ‘1.2.7’
* package encoding: UTF-8
* checking package namespace information ... OK
* checking package dependencies ...Warning: unable to access index for repository https://CRAN.R-project.org/src/contrib:
  Line starting '<!DOCTYPE html ...' is malformed!
 OK
* checking if this is a source package ... OK
* checking if there is a namespace ... OK
* checking for executable files ... OK
* checking for hidden files and directories ... OK
* checking for portable file names ... OK
* checking for sufficient/correct file permissions ... OK
* checking whether package ‘igraph’ can be installed ... OK
* checking installed package size ... NOTE
  installed size is 20.9Mb
  sub-directories of 1Mb or more:
    R      2.0Mb
    help   2.0Mb
    libs  16.4Mb
* checking package directory ... OK
* checking DESCRIPTION meta-information ... OK
* checking top-level files ... OK
* checking for left-over files ... OK
* checking index information ... OK
* checking package subdirectories ... OK
* checking R files for non-ASCII characters ... OK
* checking R files for syntax errors ... OK
* checking whether the package can be loaded ... OK
* checking whether the package can be loaded with stated dependencies ... OK
* checking whether the package can be unloaded cleanly ... OK
* checking whether the namespace can be loaded with stated dependencies ... OK
* checking whether the namespace can be unloaded cleanly ... OK
* checking loading without being on the library search path ... OK
* checking dependencies in R code ... OK
* checking S3 generic/method consistency ... OK
* checking replacement functions ... OK
* checking foreign function calls ... OK
* checking R code for possible problems ... OK
* checking Rd files ... OK
* checking Rd metadata ... OK
* checking Rd cross-references ... OK
* checking for missing documentation entries ... OK
* checking for code/documentation mismatches ... OK
* checking Rd \usage sections ... OK
* checking Rd contents ... OK
* checking for unstated dependencies in examples ... OK
* checking line endings in shell scripts ... OK
* checking line endings in C/C++/Fortran sources/headers ... OK
* checking line endings in Makefiles ... OK
* checking compilation flags in Makevars ... OK
* checking for GNU extensions in Makefiles ... OK
* checking for portable use of $(BLAS_LIBS) and $(LAPACK_LIBS) ... OK
* checking use of PKG_*FLAGS in Makefiles ... OK
* checking include directives in Makefiles ... OK
* checking compiled code ... OK
* checking examples ... OK
* checking for unstated dependencies in ‘tests’ ... OK
* checking tests ...
  Running ‘testthat.R’
 ERROR
Running the tests in ‘tests/testthat.R’ failed.
══ Failed tests ════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════
── Failure (test_layout.fr.R:12:3):  ───────────────────────────────────────────
isTRUE(all.equal(sum(l), 10.794223604849)) || isTRUE(all.equal(sum(l),
    10.7943032688805)) is not TRUE

  `actual`:   FALSE
  `expected`: TRUE
  ── Failure (test_layout.fr.R:20:3):  ───────────────────────────────────────────
  isTRUE(all.equal(sum(l), 1004.00737470853)) || isTRUE(all.equal(sum(l),
      941.472420651506)) is not TRUE

  `actual`:   FALSE
  `expected`: TRUE

  [ FAIL 2 | WARN 0 | SKIP 3 | PASS 3328 ]
  Error: Test failures
  Execution halted
```
Can you tell me how to fix it?

**Comment by ntamas** [comment]
This failure is from a test case that is supposed to smoke-test the output of a force-directed layout algorithm (Fruchterman-Reingold). There can be at least three causes of a test failure here and we need to figure out which one is the case to see whether we should just relax the test conditions or start looking for a genuine bug:

1. The layout algorithm is randomized, but we set the random seed from R before calling the layout, and igraph uses R's RNG when it runs inside R. So, in theory, the same seed should return the same layout. If this is not the case, the integration with R's RNG could be broken, but I find this highly unlikely as there are many other tests for randomized functions and they do not fail in your setup.
2. This test actually validates the result by comparing the sum of the XY coordinates in the layout with known values, with a certain tolerance. Numerical errors may have accumulated in the process to an extent that the sum diverged from the expectation; in this case there is no cause for alarm as we simply need to increase the tolerance of the test.
3. Something is different between R in CentOS and the "official R" such that they generate random numbers in a different manner. Again, I find this highly unlikely due to the same reasons outlined in 1.

So, to determine whether case 2 is at play here, can you change the `all.equal()` calls in `test_layout.fr.R` and add `tolerance=1e-4` there? The default tolerance is the square root of the machine epsilon and it might be too small in this case, although it has to be noted that we've never seen this problem in any of the platforms that we test on.

**Comment by Tom-python0121** [comment]
@ntamas I have find the package to fix it on centos8_aarch64 and centos8_x86_64:https://src.fedoraproject.org/rpms/R-igraph/blob/f31/f/0004-Increase-tolerances-to-work-on-all-arches.patch . you can see ，Can all be incorporated into the new release version?

**Comment by Tom-python0121** [comment]
the result of r-igraph@1.2.8：
![image](https://user-images.githubusercontent.com/77763324/142828660-77d49915-47e8-495b-bda5-d18e9e2661e1.png)

**Comment by ntamas** [comment]
Patch merged in 6d11778a, thanks a lot!
